core/unit_test/TestTeam.hpp

```diff
@@ -700,11 +700,11 @@ struct ScratchTeamFunctor {
   KOKKOS_INLINE_FUNCTION
   void operator()(const typename policy_type::member_type &ind,
                   value_type &update) const {
-    const shared_int_array_type scratch_ptr(ind.team_scratch(1),
+    const shared_int_array_type scratch_ptr(ind.template team_scratch<1>(),
```

Comment:
What's the deal with all the changes in the tests?
You sprinkled changes to showcase the new syntax or are these changes required?
Same questions everywhere.

Reply:
Just for showcasing/testing the changes. There are no changes in the tests required but I would add some once I get the OK on the design otherwise.

Reply:
Understood

Reply:
I think we should actually be explicit about the level whenever we know it at compile-time.